app/controllers/api/v1/dependencies_controller.rb

```diff
@@ -0,0 +1,32 @@
+class Api::V1::DependenciesController < Api::BaseController
+  skip_before_action :verify_authenticity_token
+  GEM_REQUEST_LIMIT = 250
+
+  def index
+    return render text: "Too many gems!", status: 422 if gem_names.size > GEM_REQUEST_LIMIT
+
+    deps = gem_names.each_with_object([]) do |gem_name, deps_array|
+      gem = Rubygem.includes(:versions).find_by_name!(gem_name)
+      gem_deps_hashes = gem.versions.sort_by(&:number).reverse_each.map do |version|
+        version_deps = version.dependencies.find_all { |d| d.scope == "runtime" }
+        {
+          :name => gem_name,
+          :number => version.number,
+          :platform => version.platform,
+          :dependencies => version_deps.map { |d| [d.name, d.requirements] }
+        }
+      end
+      deps_array << gem_deps_hashes
+    end
+    respond_to do |format|
+      format.json { render json: deps }
+      format.marshal { Marshal.dump(deps) }
```

Comment:
These are both going to be pretty expensive. What's the story for caching these requests?